core/domain/src/main/java/org/the_chance/honeymart/domain/usecase/ValidationSignupFieldsUseCase.kt

```diff
@@ -0,0 +1,63 @@
+package org.the_chance.honeymart.domain.usecase
+
+import org.the_chance.honeymart.domain.util.ValidationState
+import javax.inject.Inject
+
+class ValidationSignupFieldsUseCase @Inject constructor() {
+    fun validateEmail(email: String): ValidationState {
+        if (email.isBlank()) {
+            return ValidationState.BLANK_EMAIL
+        }
+        if (!isEmailMatchRegex(email)) {
+            return ValidationState.INVALID_EMAIL
+        }
+        return ValidationState.VALID_EMAIL
+    }
+
+    private fun isEmailMatchRegex(email: String): Boolean =
+        Regex("[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}").matches(email)
+
+    fun validationFullName(fullName: String): ValidationState {
+        if (fullName.isBlank()) {
+            return ValidationState.BLANK_FULL_NAME
+        }
+        if (!isFullNameMatchRegex(fullName)) {
+            return ValidationState.INVALID_FULL_NAME
+
+        }
+        return ValidationState.VALID_FULL_NAME
+    }
+
+    private fun isFullNameMatchRegex(fullName: String): Boolean =
+        Regex("^[a-zA-Z]{4,14}\\s[a-zA-Z]{4,14}$").matches(fullName)
+
+    fun validationPassword(password: String): ValidationState {
+        return when {
```

Comment:
Use **if** **else** in this case.

Reply:
both aim for the same purpose plus "when" in this state will cover all available cases.